tests/foreman/cli/test_realm.py

```diff
@@ -0,0 +1,327 @@
+# -*- encoding: utf-8 -*-
+# pylint: disable=no-self-use
+"""Test class for Realm CLI
+
+:Requirement: Realm
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: CLI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import random
+
+from fauxfactory import gen_string
+from robottelo.cleanup import capsule_cleanup
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.factory import (
+    make_location,
+    make_proxy,
+    make_realm,
+    CLIFactoryError,
+)
+from robottelo.cli.realm import Realm
+from robottelo.decorators import (
+    tier1,
+)
+from robottelo.test import CLITestCase
+
+
+class RealmTestCase(CLITestCase):
+    """Tests for Realms via Hammer CLI, must be run on QE RHEL7 Sat6.3 host"""
+
+    def _make_proxy(self, options=None):
+        """Create a Proxy and register the cleanup function"""
+        proxy = make_proxy(options=options)
+        # Add capsule to cleanup list
+        self.addCleanup(capsule_cleanup, proxy['id'])
+        return proxy
+
+    @tier1
+    def test_positive_create_delete_name(self):
+        """Proxy deletion by realm name
+
+        :id: ef3967e6-d53d-4dec-b74f-c20448d5fc6d
+
+        :expectedresults: Proxy is created and deleted
+        """
+        name = gen_string('alpha', random.randint(1, 30))
+        with self.subTest(name):
+            proxy = self._make_proxy({u'name': name})
+            self.assertEquals(proxy['name'], name)
+            realm = make_realm({'name': gen_string(
+                'alpha', random.randint(1, 30)),
+                                'realm-proxy-id': proxy['id'],
+                                'realm-type': 'Red Hat Identity Management',
+                                'locations': proxy['locations']})
+            self.assertEquals(len(Realm.list()), 1)
+            Realm.delete({'name': realm['name']})
+            self.assertEquals(len(Realm.list()), 0)
+
+    @tier1
+    def test_positive_create_delete_id(self):
+        """Proxy deletion by realm id
+
+        :id: 7c1aca0e-9724-40de-b38f-9189bdae0514
+
+        :expectedresults: Proxy is created and deleted
+        """
+        name = gen_string('alpha', random.randint(1, 30))
+        with self.subTest(name):
+            proxy = self._make_proxy({u'name': name})
+            realm = make_realm({'name': gen_string(
+                'alpha', random.randint(1, 30)),
+                                'realm-proxy-id': proxy['id'],
+                                'realm-type': 'Active Directory',
```

Comment:
Currently only `IDM` works and not `Active Directory`, this was never tested by us/QE as this is not yet supported by satellite6. Also by changing the `realm-type`, we would also invalidate the setup already done for `IDM`.